src/SprykerEco/Zed/CrefoPay/Business/Hook/Checkout/CrefoPayCheckoutPostSaveHook.php

```diff
@@ -0,0 +1,130 @@
+
+class CrefoPayCheckoutPostSaveHook implements CrefoPayCheckoutHookInterface
+{
+    protected const ERROR_TYPE_PAYMENT_FAILED = 'payment failed';
+    protected const ERROR_MESSAGE_PAYMENT_FAILED = 'Something went wrong with your payment. Try again!';
+
+    /**
+     * @var \SprykerEco\Zed\CrefoPay\Dependency\Facade\CrefoPayToCrefoPayApiFacadeInterface
+     */
+    protected $crefoPayApiFacade;
+
+    /**
+     * @var \SprykerEco\Zed\CrefoPay\Business\Hook\Checkout\Mapper\CrefoPayCheckoutHookMapperInterface
+     */
+    protected $mapper;
+
+    /**
+     * @var \SprykerEco\Zed\CrefoPay\Business\Hook\Checkout\Saver\CrefoPayCheckoutHookSaverInterface
+     */
+    protected $saver;
+
+    /**
+     * @param \SprykerEco\Zed\CrefoPay\Dependency\Facade\CrefoPayToCrefoPayApiFacadeInterface $crefoPayApiFacade
+     * @param \SprykerEco\Zed\CrefoPay\Business\Hook\Checkout\Mapper\CrefoPayCheckoutHookMapperInterface $mapper
+     * @param \SprykerEco\Zed\CrefoPay\Business\Hook\Checkout\Saver\CrefoPayCheckoutHookSaverInterface $saver
+     */
+    public function __construct(
+        CrefoPayToCrefoPayApiFacadeInterface $crefoPayApiFacade,
+        CrefoPayCheckoutHookMapperInterface $mapper,
+        CrefoPayCheckoutHookSaverInterface $saver
+    ) {
+        $this->crefoPayApiFacade = $crefoPayApiFacade;
+        $this->mapper = $mapper;
+        $this->saver = $saver;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     * @param \Generated\Shared\Transfer\CheckoutResponseTransfer $checkoutResponseTransfer
+     *
+     * @return void
+     */
+    public function execute(QuoteTransfer $quoteTransfer, CheckoutResponseTransfer $checkoutResponseTransfer): void
+    {
+        if ($quoteTransfer->getPayment()->getPaymentProvider() !== CrefoPayConfig::PROVIDER_NAME) {
+            return;
+        }
+
+        $requestTransfer = $this->mapper
+            ->mapQuoteTransferToRequestTransfer(
+                $quoteTransfer,
+                new CrefoPayApiRequestTransfer()
+            );
+
+        $responseTransfer = $this->crefoPayApiFacade->performReserveApiCall($requestTransfer);
+        $this->saver->savePaymentEntities($requestTransfer, $responseTransfer);
+
+        if (!$responseTransfer->getIsSuccess()) {
+            $this->processFailureResponse($checkoutResponseTransfer);
+
+            return;
+        }
+
+        if (!$this->isMethodWithRedirect($responseTransfer)) {
+            return;
+        }
+
+        $this->setRedirectToPaymentSystem($checkoutResponseTransfer, $responseTransfer);
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\CrefoPayApiResponseTransfer $responseTransfer
+     *
+     * @return bool
+     */
+    protected function isMethodWithRedirect(CrefoPayApiResponseTransfer $responseTransfer): bool
+    {
+        return !empty($responseTransfer->getReserveResponse()->getRedirectUrl());
```

Comment:
IMO typecasting to bool would be enough

Reply:
I think check to empty is better understandable then cast to bool type strings, null or any other types that can API send to us.